catbird/utils.py

```diff
@@ -0,0 +1,73 @@
+"""I/O utilities"""
+import json
+import subprocess
+
+def json_from_exec(exec):
+    """
+    Returns the Python objects corresponding to the MOOSE application described
+    by the json file.
+
+    Parameters
+    ----------
+    json_file : str, or Path
+        Either an open file handle, or a path to the json file. If `json` is a
+        dict, it is assumed this is a pre-parsed json object.
+
+    Returns
+    -------
+    dict
+        A dictionary of all MOOSE objects
+    """
+    json_proc = subprocess.Popen([exec, '--json'], stdout=subprocess.PIPE)
+    json_str = ''
+
+    # filter out the header and footer from the json data
+    while True:
+        line = json_proc.stdout.readline().decode()
+        if not line:
+            break
+        if '**START JSON DATA**' in line:
+            continue
+        if '**END JSON DATA**' in line:
+            continue
+
+        json_str += line
```

Comment:
I tried this out with a Cardinal executable that had a lot of deprecation warnings appear before the JSON data. This code block grabs only the content between `**START JSON DATA**` and `**END JSON DATA**`:

```python
 while json_proc.stdout.readline().decode().find('**START JSON DATA**') < 0:
        pass

    while True:
        line = json_proc.stdout.readline().decode()
        if line.find('**END JSON DATA**') >= 0:
            break
        json_str += line
```

I'll note that this is probably a pre-existing issue created by the original author 😁

Reply:
Sure happy to make that change  :)